test/tests/kernels/transient_vector_diffusion/tests

```diff
@@ -0,0 +1,21 @@
+[Tests]
+  [./transient_vector_diffusion]
+    type = 'Exodiff'
+    input = 'transient_vector_diffusion.i'
+    exodiff = 'transient_vector_diffusion_out.e'
+    requirement = 'We shall be able to reproduce the simple_transient_diffusion results with a vector compatible TimeDerivative (VectorTimeDerivative) containing a hand-coded Jacobian.'
```

Comment:
@permcody has been trying to get us to be better at how we do requirements. See his comments [here](https://github.com/idaholab/moose/pull/13944#discussion_r316912819) and [here](https://github.com/idaholab/moose/pull/13944#discussion_r316913366)

Reply:
Shortened and modified the requirement to (hopefully) better fit the guidelines.